File: Valid Parentheses.py
def match(a,b):
    if a=="(" and b==")":
        return True
    elif a=="{" and b=="}":
        return True
    elif a=="[" and b=="]":
        return True
    else:
        return False
class Solution:
    # @return a boolean
    def isValid(self, s):
        temp=[]
        l=len(s)
        if l % 2!=0:
            return False
        for i in s:
            if i=="(" or i=="[" or i=="{" :
                temp.append(i)
            else:
                if temp==[]:
                    return False
                t=temp.pop()
                if match(t,i)==False:
                    return False
        if temp==[]:
            return True
        else:
            return False


string=")}{({))[{{[}"                
s=Solution()
print s.isValid(string)

File: java/ValidParentheses.java
import java.util.Stack;
public class ValidParentheses{
	public boolean isMatch(char a, char b){
		if(a=='(' && b==')')
			return true;
		if(a=='[' && b==']')
			return true;
		if(a=='{' && b=='}')
			return true;
		return false;
	}
	public boolean isValid(String str){
		if(str.length()%2!=0)
			return false;
		Stack<Character> myStack = new Stack();
		for(int i=0;i<str.length();i++){
			char temp=str.charAt(i);
			if(temp=='(' || temp=='[' || temp=='{')
				myStack.add(temp);
			else{
				if(myStack.isEmpty())
					return false;
				else{
					char t=myStack.pop();
					if(!isMatch(t,temp))
						return false;
				}
			}
		}
		if(myStack.isEmpty())
			return true;
		else return false;
	}
	public static void main(String args[]){
		ValidParentheses vp= new ValidParentheses();
		String str="{[()[]]()}";
		System.out.println(vp.isValid(str));
		     
	}
}